src/backend/distributed/transaction/worker_transaction.c

```diff
@@ -69,6 +69,58 @@ SendCommandToWorkers(TargetWorkerSet targetWorkerSet, char *command)
 
 
 /*
+ * SendBareCommandListToWorkers sends a list of commands to a set of target
+ * workers in serial. Commands are committed immediately: new connections are
+ * always used and no transaction block is used (hence "bare"). The connections
+ * are made as the extension owner to ensure write access to the Citus metadata
+ * tables. Primarly useful for INDEX commands using CONCURRENTLY.
+ */
+void
+SendBareCommandListToWorkers(TargetWorkerSet targetWorkerSet, List *commandList)
+{
+	List *workerNodeList = WorkerNodeList();
+	ListCell *workerNodeCell = NULL;
+	char *nodeUser = CitusExtensionOwnerName();
+	ListCell *commandCell = NULL;
+
+	if (XactModificationLevel > XACT_MODIFICATION_NONE)
```

Comment:
Do we need this? PostgreSQL does not allow index concurrently in transactions anyway. Also, this error message is not visible because of PG_TRY()/PG_CATCH().

If we ditch this, we can also run `ExecuteSequentialTasksWithoutResults` before metadata sync above and reduce the window for the mx sync problem.